sources/worm.h

```diff
@@ -2,13 +2,16 @@
 
 #include <vector>
 #include <thread>
+#include <mutex>
 
 constexpr int LEFT = 0, UP = 1, RIGHT = 2, DOWN = 3;
 
 class Board;
 
 class Worm {
  protected:
+  static std::mutex mutex_;
```

Comment:
Like I said on the meeting - this is not the best place for it. Within small application we can afford one global mutex to protect the recources, but in general the mutexes should be guards standing right next to the resource that should be protected.

In this project board is the recource that all worms need access to, so I think that there should be a mutex. I think all the logic that aquires a mutex should also be implemented inside board, so that the resource protects itself rather than just moving the mutex into different file. I'd say that board itself should be thread-safe and shouldn't depend on the implementation of the threads.

Reply:
and this)